src/Microsoft.Health.Fhir.Core/Features/Search/SearchResult.cs

```diff
@@ -18,24 +20,42 @@ public class SearchResult
         /// Initializes a new instance of the <see cref="SearchResult"/> class.
         /// </summary>
         /// <param name="results">The search results.</param>
+        /// <param name="unsupportedSearchParameters">The list of unsupported search parameters.</param>
         /// <param name="continuationToken">The continuation token.</param>
-        public SearchResult(IEnumerable<ResourceWrapper> results, string continuationToken)
+        public SearchResult(IEnumerable<ResourceWrapper> results, IReadOnlyList<Tuple<string, string>> unsupportedSearchParameters, string continuationToken)
         {
             EnsureArg.IsNotNull(results, nameof(results));
+            EnsureArg.IsNotNull(unsupportedSearchParameters, nameof(unsupportedSearchParameters));
 
             Results = results;
+            UnsupportedSearchParameters = unsupportedSearchParameters;
             ContinuationToken = continuationToken;
         }
 
+        public SearchResult(int? totalCount, IReadOnlyList<Tuple<string, string>> unsupportedSearchParameters)
+        {
+            EnsureArg.IsNotNull(totalCount, nameof(totalCount));
```

Comment:
I think we should either make totalCount non-nullable or remove the NotNull check.